framework/decode/file_processor.cpp

```diff
@@ -485,9 +485,12 @@ bool FileProcessor::ReadCompressedParameterBuffer(size_t  compressed_buffer_size
 
 bool FileProcessor::ReadBytes(void* buffer, size_t buffer_size)
 {
-    size_t bytes_read = util::platform::FileRead(buffer, 1, buffer_size, file_descriptor_);
-    bytes_read_ += bytes_read;
-    return (bytes_read == buffer_size);
+    if (util::platform::FileRead(buffer, buffer_size, file_descriptor_))
```

Comment:
This changes the behavior of this function - if FileRead read short, previously `bytes_read` would be updated even though it reported `false`.  With this change, `bytes_read` won't be updated.  So if a caller was okay with FileProcessor::ReadBytes making a short read for some reason and expecting `bytes_read` to be updated, that won't work with this change.

Reply:
FileRead will now no longer ever read short, however. When called, it either fails or does a full read. This is guaranteed by the full item call to fread (ie calling it with bytes item size and 1 item count). Same for FileWrite. This greatly simplifies calling code.